sbncode/CAFMaker/CAFMaker_module.cc

```diff
@@ -443,12 +446,28 @@ void CAFMaker::beginRun(art::Run& run) {
 void CAFMaker::beginSubRun(art::SubRun& sr) {
   // get the POT
   // get POT information
-  art::Handle<sumdata::POTSummary> pot_handle;
-  sr.getByLabel("generator", pot_handle);
-
-  if (pot_handle.isValid()) {
-    fSubRunPOT = pot_handle->totgoodpot;
-    fTotalPOT += fSubRunPOT;
+  if(fIsRealData)
+  {
+    fBNBInfo.clear();
+    fSubRunPOT = 0;
+    art::Handle<std::vector<sbn::BNBSpillInfo> > bnb_spill;
+    sr.getByLabel(fParams.BNBPOTDataLabel(), bnb_spill);
+    if (bnb_spill.isValid()) {
+      std::vector<art::Ptr<sbn::BNBSpillInfo> >  bnb_spill_info;
+      art::fill_ptr_vector(bnb_spill_info, bnb_spill);
+      FillExposure(bnb_spill_info, fBNBInfo, fSubRunPOT);
+      fTotalPOT += fSubRunPOT;
+    }
+  }
+  else
+  {
+    art::Handle<sumdata::POTSummary> pot_handle;
+    sr.getByLabel(fParams.GenLabel(), pot_handle);
+    
+    if (pot_handle.isValid()) {
```

Comment:
Fun with C++ (not particularly endorsing this change):
```suggestion
    if (auto pot_handle = sr.getHandle<sumdata::POTSummary>(fParams.GenLabel())) {
```